python/ray/util/sgd/v2/callbacks/logging.py

```diff
@@ -87,3 +121,125 @@ def handle_result(self, results: Optional[List[Dict]]):
             f.seek(0)
             json.dump(
                 loaded_results + [results_to_log], f, cls=SafeFallbackEncoder)
+
+
+class SGDSingleFileSingleWorkerLoggingCallback(
+        SGDSingleFileLoggingCallback, metaclass=abc.ABCMeta):
+    """Abstract SGD logging callback class.
+
+    Allows only for single-worker logging.
+
+    Args:
+        logdir (Optional[str]): Path to directory where the results file
+            should be. If None, will be set by the Trainer.
+        filename (Optional[str]): Filename in logdir to save results to.
+        workers_to_log (int): Worker index to log. By default, will log the
+            worker with index 0.
+    """
+
+    # it's still workers_to_log, not worker, for easier switching
+    # this is defined again for different type hints
+    def __init__(self,
+                 logdir: Optional[str] = None,
+                 filename: Optional[str] = None,
+                 workers_to_log: int = 0) -> None:
```

Comment:
Would `worker_to_log` be a better name here instead of `workers_to_log`, since it's only 1 worker?

Reply:
The argument name is left as-is to maintain compatibility with the parent class.

Reply:
Isn't it fine though if this class has different args than the parent class?

Reply:
It wouldn't break anything, though it seems off to me from an OOP design standpoint. It's my belief that every subclass should have the same arguments and attributes as the parent class, plus any extra ones - but it should not remove or modify the args/attrs of the parent. That being said `worker_to_log` would be more in line with what's actually happening. Let me know what you think, I'm fine with either.